File: tests/human/x/hs/group_items_iteration.hs
module Main where

import qualified Data.Map.Strict as M
import Data.List (sortOn)

data Item = Item { tag :: String, val :: Int }

items :: [Item]
items = [ Item "a" 1, Item "a" 2, Item "b" 3 ]

groups :: [(String, [Item])]
groups = M.toList $ M.fromListWith (++) [(tag i, [i]) | i <- items]

result :: [(String, Int)]
result = [ (t, sum [val x | x <- xs]) | (t,xs) <- groups ]

sorted :: [(String, Int)]
sorted = sortOn fst result

main :: IO ()
main = print [ M.fromList [("tag", t), ("total", s)] | (t,s) <- sorted ]

File: tests/human/x/python/group_items_iteration.py
data = [
    {"tag": "a", "val": 1},
    {"tag": "a", "val": 2},
    {"tag": "b", "val": 3},
]

# Group data by tag
groups = {}
for d in data:
    groups.setdefault(d["tag"], []).append(d)

tmp = []
for tag, items in groups.items():
    total = sum(x["val"] for x in items)
    tmp.append({"tag": tag, "total": total})

result = sorted(tmp, key=lambda r: r["tag"])
print(result)
